pkg/workerapi/workerserver/server.go

```diff
@@ -48,24 +49,24 @@ func (s *server) Serve(bindAddress string) error {
 		cmux.HTTP2MatchHeaderFieldSendSettings("content-type", "application/grpc"))
 	httpListener := mux.Match(cmux.Any())
 
-	logIfErrored := func(protocol string, f func() error) {
-		if err := f(); err != nil {
+	logIfErrored := func(protocol string, err error) {
+		if err != nil && !errors.Is(err, net.ErrClosed) && !errors.Is(err, cmux.ErrListenerClosed) {
 			log.Error().WithError(err).Messagef("Error during serving %s.", protocol)
 		}
 	}
 
-	go logIfErrored("gRPC", func() error { return serveGRPC(s.grpc, grpcListener) })
-	go logIfErrored("HTTP", func() error { return serveHTTP(s.rest, httpListener) })
-
-	return mux.Serve()
+	go logIfErrored("mux", mux.Serve())
+	go logIfErrored("HTTP", serveHTTP(s, s.rest, httpListener))
+	return serveGRPC(s.grpc, grpcListener)
 }
 
-// Close closes the server. No attempt to finish active requests is made.
+// Close closes the server.
 //
-// Any active gRPC connections will be notified by connection errors.
+// Tries to gracefully stop gRPC requests and connections.
+// Abruptly stops active HTTP requests.
 func (s *server) Close() error {
 	if s.grpc != nil && s.grpc.grpc != nil {
-		s.grpc.grpc.Stop()
+		s.grpc.grpc.GracefulStop()
```

Comment:
Would be nice to have a timeout where it calls `s.grpc.grpc.Stop` if `GracefulStop` takes too long.

`GracefulStop` will actually wait for all streams to close themselves. If we never close the streams via our code ourselves, then `GracefulStop` will hault forever.

Maybe something like this would suffice?

```suggestion
		const timeout = 5 * time.Second
		log.Debug().WithDuration("timeout", timeout).
			Message("Attempting to shut down gracefully.")

		timer := time.AfterFunc(timeout, func() {
			log.Debug().Message("Timeout exceeded. Shutting down immediately.")
			s.grpc.grpc.Stop()
		})
		s.grpc.grpc.GracefulStop()
		timer.Stop()
```

(also need to add `"time"` to imports)